dm/dm/ctl/master/task_operate.go

```diff
@@ -0,0 +1,179 @@
+	argLen := len(cmd.Flags().Args())
+	if argLen == 0 {
+		// may want to operate tasks bound to a source
+		return sourceTaskOperateFunc(cmd, operateFunc)
+	} else if argLen > 1 {
+		// can pass at most one task-name/task-conf
+		cmd.SetOut(os.Stdout)
+		common.PrintCmdUsage(cmd)
+		return errors.New("please check output to see error")
+	}
+
+	taskName := common.GetTaskNameFromArgOrFile(cmd.Flags().Arg(0))
+	sources, err := common.GetSourceArgs(cmd)
+	if err != nil {
+		return err
+	}
+
+	// check task is exist
+	notExist, err := isTaskNotExist(taskName)
+	if notExist || err != nil {
+		return err
+	}
+
+	common.PrettyPrintInterface(operateFunc(cmd, taskName, sources))
+	return nil
+}
+
+func addSourceTaskOperateFlags(cmd *cobra.Command) {
+	// control workload to dm-cluster for sources with large number of tasks.
+	cmd.Flags().Int(operateBatchSizeFlag, defaultOperateBatchSize, "batch size when operating all (sub)tasks bound to a source")
+}
+
+func parseSourceTaskOperateParams(cmd *cobra.Command) (string, int, error) {
+	sources, err := common.GetSourceArgs(cmd)
+	if err != nil {
+		return "", 0, err
+	}
+	if len(sources) == 0 {
+		common.PrintLinesf(`must give one source-name when task-name/task-conf is not specified`)
+		return "", 0, errors.New("missing source")
+	} else if len(sources) > 1 {
+		common.PrintLinesf(`can give only one source-name when task-name/task-conf is not specified`)
+		return "", 0, errors.New("too many source")
+	}
+	batchSize, err := cmd.Flags().GetInt(batchSizeFlag)
+	if err != nil {
+		common.PrintLinesf("error in parse `--" + batchSizeFlag + "`")
+		return "", 0, err
+	}
+	return sources[0], batchSize, nil
+}
+
+func sourceTaskOperateFunc(cmd *cobra.Command,
+	operateFunc func(cmd *cobra.Command, taskName string, sources []string) *taskOperateResult,
+) error {
+	source, batchSize, err := parseSourceTaskOperateParams(cmd)
+	if err != nil {
+		cmd.SetOut(os.Stdout)
+		common.PrintCmdUsage(cmd)
+		return errors.New("please check output to see error")
+	}
+
+	sources := []string{source}
+	sourceList := openapi.SourceNameList(sources)
+	ctx, cancel := context.WithTimeout(context.Background(), common.GlobalConfig().RPCTimeout)
+	defer cancel()
+
+	listReq := &openapi.DMAPIGetTaskListParams{SourceNameList: &sourceList}
+	resp := &openapi.GetTaskListResponse{}
+	params := []interface{}{listReq}
+	sendErr := common.SendOpenapiRequest(ctx, "DMAPIGetTaskList", params, http.StatusOK, resp)
+	if sendErr != nil {
+		common.PrintLinesf("cannot query status of source: %v", sources)
+		return errors.Trace(sendErr)
+	}
+
+	if resp.Total == 0 {
+		common.PrettyPrintOpenapiResp(false, fmt.Sprintf("there is no tasks of source: %v", sources))
+		return nil
+	}
+	result := batchTaskOperate(cmd, batchSize, sources, resp.Data, operateFunc)
+	common.PrettyPrintInterface(result)
+	return nil
+}
+
+func batchTaskOperate(cmd *cobra.Command, batchSize int, sources []string, tasks []openapi.Task,
```

Comment:
it seem that don't need batch request here? sotp task api support suourceNameList param

Reply:
`batchTaskOperate` is called by `task stop -s xxx` which will stop all task about the source, and `task stop xx -s xx,xx` will just call api support suourceNameList param.

Reply:
<img width="1042" alt="Screen Shot 2022-06-24 at 09 25 16" src="https://user-images.githubusercontent.com/24697284/175441417-2c292aac-68ee-4402-a252-7f2307721bc2.png">


i think all task cmds must have a `task-name` arg, if user want stop all subtasks about a source, he can use `api/v1/source/source-name/disable` instead